### PR TITLE
Add trrosetta dataset

### PR DESCRIPTION
**Pull request by blazingsiyan**
include (id, seq, msa, coords, angles, dist, mask, msa_mask)

- download and extract trrosetta dataset from s3
- read msa from a3m file
- read pdb file using `prody`, process using `sidechainnet`
- calculate bucketed distance based on ca or cb
- optional pytorch lightning data module

caveats
- the performance is not good :disappointed:
- not fully tested :hushed:

### CODE REVIEW

**Comment by hypnopump** [comment]
Oh this looks very nice! Very cool functionality! I think this covers until CASP13 or more? If so, this is great: more data!
Also, are the MSAs in trRosetta processed somehow?

What do you mean by "the performance is not good"? Idk if we should leave it as a branch until we can fix the possible errors/caveats you mentioned or merge it into main... I'd like to hear @lucidrains ' opinion as well here

**Comment by lucidrains** [comment]
@blazingsiyan this is tremendous! thank you so much :) this will help polish up the trunk. there's no way we can't beat trRosetta, at the very least :)

**Comment by blazingsiyan** [comment]
> Oh this looks very nice! Very cool functionality! I think this covers until CASP13 or more? If so, this is great: more data!
> Also, are the MSAs in trRosetta processed somehow?
> 
> What do you mean by "the performance is not good"? Idk if we should leave it as a branch until we can fix the possible errors/caveats you mentioned or merge it into main... I'd like to hear @lucidrains ' opinion as well here

Sorry for the late response :stuck_out_tongue:

By "performance" I mean the read/cache/collate cost of the dataloader and I think there is room for improvement.

The MSAs are the same as in the original paper. 

Ideally, we should have the sequence DB, then use HHblits/JackHMMER to generate the MSAs

but there will be terabytes of data and will take several weeks for searching.

Before achieving that, I think we should use some existing datasets to validate our model :blush:

**Comment by hypnopump** [comment]
Oh yes! I can help with performance, dont worry! 

As for the MSA... we have the MSA data up until CASP12 (from proteinNet), so we can do several tests with that for now (although it would be ideal if we can have the newest MSA data once we have the model ready for training)
